### ISSUE TITLE
nalgebra feature not working as advertised in README.md

### ISSUE THREAD

**Issue by DriesCruyskens** [u1roh/kd-tree#8]
When following [your official docs](https://github.com/u1roh/kd-tree#nalgebra-feature), using the nalgebra feature does not work as advertised. The following error is given:

```
the trait bound `OPoint<i32, Const<3_usize>>: KdPoint` is not satisfied
the trait `KdPoint` is not implemented for `OPoint<i32, Const<3_usize>>`
```

So I'm not sure if the feature is broken or the docs are wrong.
I made an MRE to demonstrate (but it is basically the docs copied and pasted into a project).

## Minimal reproducible example
<https://github.com/DriesCruyskens/test-kd-tree>

**Comment by u1roh** [comment]
@DriesCruyskens Sorry for may lazy response.
It doesn't compile because the nalgebra' version you use is different from the version of kd-tree.
I will upgrade and release it.

**Comment by u1roh** [comment]
I've released v0.4.2.
https://crates.io/crates/kd-tree/0.4.2

The latest version uses nalgebra 0.31.1.
```
nalgebra = { version = "0.31.1", optional = true }
```
